HPXMLtoOpenStudio/tests/test_schedules.rb

```diff
@@ -207,36 +248,36 @@ def test_stochastic_vacancy_schedules
-    assert_in_epsilon(887 * occupied_ratio, sf.annual_equivalent_full_load_hrs(col_name: SchedulesFile::ColumnHotWaterFixtures, schedules: sf.tmp_schedules), 0.1)
+    unavailable_month_hrs = { 0 => 31.0 * 24.0, 11 => 31.0 * 24.0 }
+
+    assert_in_epsilon(6689 * get_available_hrs_ratio(unavailable_month_hrs), sf.annual_equivalent_full_load_hrs(col_name: SchedulesFile::ColumnOccupants, schedules: sf.tmp_schedules), @stochastic_sch_tol)
+    assert_in_epsilon(2086 * get_available_hrs_ratio(unavailable_month_hrs, Schedule.LightingInteriorMonthlyMultipliers), sf.annual_equivalent_full_load_hrs(col_name: SchedulesFile::ColumnLightingInterior, schedules: sf.tmp_schedules), @stochastic_sch_tol)
+    assert_in_epsilon(2086 * get_available_hrs_ratio(unavailable_month_hrs, Schedule.LightingInteriorMonthlyMultipliers), sf.annual_equivalent_full_load_hrs(col_name: SchedulesFile::ColumnLightingGarage, schedules: sf.tmp_schedules), @stochastic_sch_tol)
+    assert_in_epsilon(2763 * get_available_hrs_ratio(unavailable_month_hrs, Schedule.LightingExteriorMonthlyMultipliers), get_annual_equivalent_full_load_hrs(model, Constants.ObjectNameLightingExterior + ' schedule'), @stochastic_sch_tol)
+    assert_in_epsilon(6673, get_annual_equivalent_full_load_hrs(model, Constants.ObjectNameRefrigerator + ' schedule'), @tight_tol)
+    assert_in_epsilon(534 * get_available_hrs_ratio(unavailable_month_hrs), sf.annual_equivalent_full_load_hrs(col_name: SchedulesFile::ColumnCookingRange, schedules: sf.tmp_schedules), @stochastic_sch_tol)
+    assert_in_epsilon(213 * get_available_hrs_ratio(unavailable_month_hrs), sf.annual_equivalent_full_load_hrs(col_name: SchedulesFile::ColumnDishwasher, schedules: sf.tmp_schedules), @stochastic_sch_tol)
+    assert_in_epsilon(134 * get_available_hrs_ratio(unavailable_month_hrs), sf.annual_equivalent_full_load_hrs(col_name: SchedulesFile::ColumnClothesWasher, schedules: sf.tmp_schedules), @stochastic_sch_tol)
+    assert_in_epsilon(151 * get_available_hrs_ratio(unavailable_month_hrs), sf.annual_equivalent_full_load_hrs(col_name: SchedulesFile::ColumnClothesDryer, schedules: sf.tmp_schedules), @stochastic_sch_tol)
+    assert_in_epsilon(3250 * get_available_hrs_ratio(unavailable_month_hrs), sf.annual_equivalent_full_load_hrs(col_name: SchedulesFile::ColumnCeilingFan, schedules: sf.tmp_schedules), @stochastic_sch_tol)
+    assert_in_epsilon(4840 * get_available_hrs_ratio(unavailable_month_hrs), sf.annual_equivalent_full_load_hrs(col_name: SchedulesFile::ColumnPlugLoadsOther, schedules: sf.tmp_schedules), @stochastic_sch_tol)
+    assert_in_epsilon(4840 * get_available_hrs_ratio(unavailable_month_hrs), sf.annual_equivalent_full_load_hrs(col_name: SchedulesFile::ColumnPlugLoadsTV, schedules: sf.tmp_schedules), @stochastic_sch_tol)
+    assert_in_epsilon(273 * get_available_hrs_ratio(unavailable_month_hrs), sf.annual_equivalent_full_load_hrs(col_name: SchedulesFile::ColumnHotWaterDishwasher, schedules: sf.tmp_schedules), 0.1) # FIXME
```

Comment:
I don't totally understand / know enough about the stochastic schedule generation, to explain why we can't estimate full load hrs (w/ vacancy or power outage) more accurately? Hence, the few FIXMEs with higher tolerance values here.

Reply:
It doesn't seem like knowledge of how the schedules are generated is necessarily needed? If you have an 8760 schedule, stochastic or not, it should be reasonably straightforward to calculate the full load hours when an unavailable period is applied. Either the test is not correct or the implementation is not correct.

Reply:
Yeah, for calculating actual full load hours I can use `sf.hours.annual_equivalent_full_load_hrs(..., schedules: sf.tmp_schedules)`. I'm trying to come up with a clever way to get expected full load hours.